### PR TITLE
New Resolver

### PR DESCRIPTION
**Pull request by lukkie112**
New Resolver

### CODE REVIEW

File: lib/resolveurl/plugins/upstream.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from __generic_resolver__ import ResolveUrl
```
Comment (host505) [inline lib/resolveurl/plugins/upstream.py]: `from __resolve_generic__ import ResolveUrl`

File: lib/resolveurl/plugins/supervideo.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+from lib import helpers, jsunpack
+from ResolveUrl import common
+from ResolveUrl.resolver import ResolveUrl, ResolverError
```
Comment (host505) [inline lib/resolveurl/plugins/supervideo.py]:
```
from resolveurl import common
from resolveurl.resolver import ResolveUrl, ResolverError
```

File: lib/resolveurl/plugins/prostream.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+from lib import helpers, jsunpack
+from ResolveUrl import common
+from ResolveUrl.resolver import ResolveUrl, ResolverError
```
Comment (host505) [inline lib/resolveurl/plugins/prostream.py]:
```
from resolveurl import common
from resolveurl.resolver import ResolveUrl, ResolverError
```

**Review by host505 — COMMENTED** [review]
Apart from import errors all 3 resolvers seem to work on my end.

File: lib/resolveurl/plugins/prostream.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+from lib import helpers, jsunpack
+from ResolveUrl import common
```
Comment (host505) [inline lib/resolveurl/plugins/prostream.py:20]: import error
```suggestion
from resolveurl import common
```

File: lib/resolveurl/plugins/prostream.py (hunk 3 above)
Comment (host505) [inline lib/resolveurl/plugins/prostream.py:21]: import error
```suggestion
from resolveurl.resolver import ResolveUrl, ResolverError
```

File: lib/resolveurl/plugins/supervideo.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+from lib import helpers, jsunpack
+from ResolveUrl import common
```
Comment (host505) [inline lib/resolveurl/plugins/supervideo.py:20]: import error
```suggestion
from resolveurl import common
```

File: lib/resolveurl/plugins/supervideo.py (hunk 2 above)
Comment (host505) [inline lib/resolveurl/plugins/supervideo.py:21]: import error
```suggestion
from resolveurl.resolver import ResolveUrl, ResolverError
```

File: lib/resolveurl/plugins/upstream.py (hunk 1 above)
Comment (host505) [inline lib/resolveurl/plugins/upstream.py:18]: import error
```suggestion
from __resolve_generic__ import ResolveUrl
```

**Comment by jsergio123** [comment]
@lukkie112 please fix the import errors as @host505 pointed out.

**Comment by lukkie112** [comment]
Fixed the Import problem with new PullRequest #263

**Pull request closed without merging**